### ISSUE TITLE
build without libGoogleAnalytics.jar and GoogleAdMobAdsSdk-6.0.1.jar

### ISSUE THREAD

**Issue by GoogleCodeExporter** [baluragala/gtalksms#293]
```
Hello,

The repo currently include two dependencies libGoogleAnalytics.jar and 
GoogleAdMobAdsSdk.*.jar . Would it be possible to offer a build that would 
disable any code based on those libraries?  That would make it possible to 
include your application in the F-Droid (100% Free and Open Source Software) 
repository.

Those libs prevented the app from being added to the repository : 
http://f-droid.org/forums/topic/gtalksms/
```

Original issue reported on code.google.com by `amu...@free.fr` on 5 Sep 2012 at 8:08

**Comment by GoogleCodeExporter** [comment]
```
First let me say that I am a big fan of the f-droid project. :-)

The Google Analytics lib was a orphan anyway. And regarding AdMob we first have 
to discuss this internally if we are willing to abstain from the income 
generated by Ads. How do this other projects handle? Is there a FOSS Ad lib for 
Android somewhere? Can we run a patch every time a new release is done on 
f-droid that removes the references to the AdMob lib?
```

Original comment by `fschm...@gmail.com` on 6 Sep 2012 at 8:05

**Comment by GoogleCodeExporter** [comment]
```
F-Droid can easily run a patch on builds to remove the AdMob stuff, if you 
wanted to leave it in for your own builds, so that Google Play releases get 
ads. The only problem with this is it might delay updates, as we'd have to 
check and possibly amend the patch each time you released. Another way to 
handle it might be to have a build flag set.

As to how other projects handle it, there are various ways. The majority don't 
have ads at all, so this doesn't come up, but there are some that have ads in 
the Market version, and not in the F-Droid version. There are also some apps 
that are in F-Droid, but paid for on Google Play. I'm not aware of a FOSS ad 
library.

In general, I would hope that F-Droid users are likely to donate if they find 
something useful to them. I can't offer much evidence of that being true, I 
just hope it is, and we encourage it by including donate links (to a dedicated 
web page if you have one, and also to a Flattr account) for each app on our web 
site, and also directly in the F-Droid client app.

Actually, one small piece of evidence - there are no ads on the F-Droid site, 
or in the app. We rely entirely on people donating because they want to, and 
they do.

The above is speaking as the founder and main developer of F-Droid. Speaking 
from a purely personal perspective, I'd never use anything with ads in, or 
anything that I don't have the full source code for. I'd *always* donate to 
something if I used it more than a couple of times. If everyone was like me, 
you wouldn't be abstaining from income, you'd be gaining it. Is everyone like 
me? I doubt it, but I don't know!

```

Original comment by `ciaran.g...@gmail.com` on 12 Sep 2012 at 6:31

**Comment by GoogleCodeExporter** [comment]
```
We are going to remove the admob lib :)
```

Original comment by `fschm...@gmail.com` on 4 Jan 2013 at 4:26
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision de97abfa6fa1.
```

Original comment by `fschm...@gmail.com` on 6 Jan 2013 at 5:16
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Fixed with 4.3 release
```

Original comment by `fschm...@gmail.com` on 26 Jan 2013 at 5:45
- Changed state: **Deployed**
